homeassistant/components/switch/__init__.py

```diff
@@ -68,9 +73,40 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     )
     await component.async_setup(config)
 
-    component.async_register_entity_service(SERVICE_TURN_OFF, {}, "async_turn_off")
-    component.async_register_entity_service(SERVICE_TURN_ON, {}, "async_turn_on")
-    component.async_register_entity_service(SERVICE_TOGGLE, {}, "async_toggle")
+    async def async_handle_switch_off_service(
+        switch: SwitchEntity, call: ServiceCall
+    ) -> None:
+        """Handle turning off a light."""
+        # pylint: disable-next=protected-access
+        switch._context_filter = state_filter(STATE_OFF)
+        await switch.async_turn_off()
```

Comment:
There is a bit of a race here if you end up calling multiple calls but I'm not sure we can do anything about that.

Reply:
Agreed there's a race, but the race is not introduced by this PR, the entity already only supports a single context which is set by the newest service call.